src/NuGet.Core/NuGet.Configuration/Settings/Items/CredentialsItem.cs

```diff
@@ -149,12 +140,13 @@ public CredentialsItem(string name, string username, string password, bool isPas
         internal CredentialsItem(XElement element, SettingsFile origin)
             : base(element, origin)
         {
-            ElementName = element.Name.LocalName;
+            // ElementName is read from XML file, so it must be decoded.
+            ElementName = XmlConvert.DecodeName(element.Name.LocalName);
 
             var elementDescendants = element.Elements();
             var countOfDescendants = elementDescendants.Count();
 
-            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin) as AddItem).Where(i => i != null);
+            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin) as AddItem).Where(i => i != null).Cast<AddItem>();
```

Comment:
While you're here we might as well change this to use `OfType<T>()` instead

```suggestion
            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin)).OfType<AddItem>();
```

Reply:
This comment made me realize my PR has a bug.

The old code could theoretically end up with nulls in the enumeration, and then when it's used, throw a NullReferenceException.

Your suggestion, and my original PR, would silently ignore the bad values, therefore changing product behaviour.

What I'm changing locally, and will push soon, is changing the `as AddItem` to a `(AddItem)` cast, so that an exception will be thrown (so similar to the original code), but it'll be an `InvalidCastException` which will tell us the actual type of the object that couldn't be cast.